### PR TITLE
Use new search API in bookEditor

### PR DESCRIPTION
**Pull request by R1kM**
This PR uses the recently introduced searchBar inside of the bookEditor; this also allows us to have more details about the search results, including for instance the deviser of a set.

Note, there is some duplication for clickable_row compared to what currently exists in AnyResultNewAPI. This is because the existing clickable row hardcodes a redirection (to an url passed in argument) for the on_click action. For the book editor, we actually need an actual action, which is to add the selected set to the current editor.

### CODE REVIEW

**Comment by Niols** [comment]
_Considering the commit history, this PR should be squashed._

> This PR uses the recently introduced searchBar inside of the bookEditor; this also allows us to have more details about the search results, including for instance the deviser of a set.

Pretty cool! Will review in a moment.

> Note, there is some duplication for clickable_row compared to what currently exists in AnyResultNewAPI. This is because the existing clickable row hardcodes a redirection (to an url passed in argument) for the on_click action. For the book editor, we actually need an actual action, which is to add the selected set to the current editor.

I am not at all convinced by the current `clickable_row`. It also lives in an odd place. I suppose we should just make it a component (albeit a tiny one) and use it everywhere. Not necessarily the scope of this PR though.

File: src/client/views/book/bookEditorInterface.ml [hunk 1]
```diff
@@ -109,25 +130,19 @@ let create ?on_save page =
...
-      page
+    let open Dancelor_client_html in
+    To_dom.of_div @@ div [
+      Dancelor_client_components.SearchBar.make
+        ~placeholder:"Add set (Magic Search)"
+        ~search
+        ~make_result:(make_set_result editor page)
+        ~max_results:10
+        ~on_enter:(fun search_text ->
+            Dom_html.window##.location##.href := js PageRouter.(path (Search (Some search_text))))
```
Comment (Niols) [inline src/client/views/book/bookEditorInterface.ml:141]: This last `on_enter` will trigger a redirection to the search page if we press enter. Is that really what we want? I feel we'd rather just not have an action on enter at all (and no “Press enter for more results” message from the bar). Is that correct or am I missing something?

**Review by Niols — COMMENTED** [review]
Very very neat improvement on the current situation and rather surgical intervention; I like it! Just one question.

File: src/client/views/book/bookEditorInterface.ml (hunk 1 above)
Comment (R1kM) [reply src/client/views/book/bookEditorInterface.ml]: Very good point, done in [3a41135](https://github.com/paris-branch/dancelor/pull/252/commits/3a4113543192f65b873c0d8e1bb0cdc17f144950)

**Pull request merged**